src/query.rs

```diff
@@ -195,6 +195,12 @@ impl Query {
         self.client.add_option(name, value);
         self
     }
+
+    pub fn with_param(self, name: &str, value: impl Serialize) -> Result<Self, String> {
```

Comment:
I'm still trying to figure out the naming here. This one is consistent with `with_option`, but that name emphasizes the connection with `Client::with_option`.

For instance, `bind` mimics sqlx [bind](https://docs.rs/sqlx/latest/sqlx/query/struct.Query.html#method.bind).
A simple `param()` would be more concise. Yep, it's inconsistent for now, but once `Client::builder()` is introduced, we can have `fn option()` instead of `with_option`.

Reply:
Shouldn't it be `with_param` on client & `param` on `Client::builder` then?

Reply:
> @loyd, I'd like to merge this one. WDYT?

@slvrtrn, this question is still unresolved